### PR TITLE
Adds fasthttp handler

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds the FastHttp implementation which is needed by dapr.

Depends on #4

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ golangci_lint := github.com/golangci/golangci-lint/cmd/golangci-lint@v1.49.0
 .PHONY: test
 test:
 	@go test -v ./...
+	@(cd handler/fasthttp; go test -v ./...)
```
Comment (codefromthecrypt) [inline Makefile:7]: adrian acks that this is not the neatest way, but still in POC mode and confident we an make make pretty

File: handler/fasthttp/middleware.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	ctx.SetUserValue("next", w.next)
+	if err := w.guest.Handle(ctx); err != nil {
+		ctx.Error(err.Error(), fasthttp.StatusInternalServerError)
+	}
+}
+
+// Close implements api.Closer
+func (w *guest) Close(ctx context.Context) error {
+	return w.guest.Close(ctx)
+}
```
Comment (codefromthecrypt) [inline handler/fasthttp/middleware.go:102]: this is one of those "screenshot quick" moments.. basically I tried to refactor until implementation was about 100 lines. I know that's never going to last, but hey it is almost the case now 😎

File: handler/fasthttp/middleware.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+type guest struct {
+	guest *internalhandler.Guest
+	next  fasthttp.RequestHandler
+}
+
+// Handle implements RequestHandler.Handle
+func (w *guest) Handle(ctx *fasthttp.RequestCtx) {
+	ctx.SetUserValue("next", w.next)
+	if err := w.guest.Handle(ctx); err != nil {
```
Comment (codefromthecrypt) [inline handler/fasthttp/middleware.go:94]: the neat subtle thing about fasthttp vs net/http is that the ctx actually implements context.Context. This means we don't really have to wrap anything and propagate it, like we do in the former.

**Comment by codefromthecrypt** [comment]
rebased to depend on #4 as that centralizes the ABI changes

**Review by anuraaga — APPROVED** [review]
Quite cool seeing the same wasm used in both net and fasthttp

**Pull request merged**